### PR TITLE
MGDAPI-3084 ensure binaryData from CM in openshif-marketplace treated well

### PR DESCRIPTION
**Pull request by makslion**
# Issue link
[MGDAPI-3084](https://issues.redhat.com/browse/MGDAPI-3084)

# What
On the 4.9+ cluster, CSV stored in config map under the `openshift-marketplace` namespace resides within `binaryData` field instead of `data`. With the old code it meant that we could find CSV and was defaulting to the default (is service affecting). With this change, we would be able to fetch binary data to make the logic about auto-approval of non-service affecting upgrades work. 

# Verification steps
To verify this you would need to simulate an upgrade. I will provide you with indexes for installations. They all are using the same RHOAM image that was built from this PR with addition of the `managed-api-service` subscription [here](https://github.com/integr8ly/integreatly-operator/blob/master/controllers/subscription/subscription_controller.go#L41-L56) 
Indexes to be used: 
* 1.16.0 - fresh install `quay.io/mvavilov/rhoam-index:1.16.0`
* 1.17.0 - replaces 1.16.0 and **is not** service-affecting (will be approved automatically) `quay.io/mvavilov/rhoam-index:1.17.0`
* 1.18.0 - replaces 1.17.0 and **is** service-affecting (will not be approved automatically)  `quay.io/mvavilov/rhoam-index:1.18.0`

1. Obtain a 4.9+ cluster
2. Install RHOAM 1.16.0 via OLM
3. When `status.stage` == `complete` point `CatalgueSource` to 1.17.0
4. Watch an install plan being created and you could watch the RHOAM pod logs to see the moment it gets to the subscription reconciles and approves the new install plan 
5. After an upgrade point `CatalogueSource` to 1.18.0. RHMI pod should go  through the subscription reconciler and the install plan should not be approved 

> There is also a new unit test to cover added logic but if prow passes it all grand

### CODE REVIEW

File: controllers/subscription/csvlocator/csvlocator.go [hunk 1]
```diff
@@ -67,44 +71,87 @@ func (l *ConfigMapCSVLocator) GetCSV(ctx context.Context, client k8sclient.Clien
...
 			Name:      ref.Name,
 			Namespace: ref.Namespace,
 		}, csvConfigMap); err != nil {
 			return csv, fmt.Errorf("error retrieving ConfigMap %s/%s: %v", ref.Namespace, ref.Name, err)
 		}
 
 		// The ConfigMap may contain other manifests other than the CSV. Iterate
-		// through the content and skip the ones that have a kind other than
+		// through the data and skip the ones that have a kind other than
 		// ClusterSeerviceVersion
```
Comment (Boomatang) [inline controllers/subscription/csvlocator/csvlocator.go:83]: Can we add a note that this is required for openshift 4.8 and put a similar note for the 4.9 block

**Review by Boomatang — COMMENTED** [review]
Generally the code looks ok. The update to comment strings is all I would like to see. Currently working on the verification steps.

File: controllers/subscription/csvlocator/csvlocator.go (hunk 1 above)
Comment (makslion) [reply controllers/subscription/csvlocator/csvlocator.go]: done

**Comment by Boomatang** [comment]
I have done the verification steps on both a 4.8 and 4.9 cluster. Service and Non Service affecting upgrades worked as expected.

/lgtm

**Pull request merged**